Fix prometheus role nfs

zgalor opened the pull request:
Allow for external or internal nfs.
Use facts as used for logging and metrics.
Also updated alertmanager image default tag to a stable version

zgalor commented:
@ewolinetz I am still testing this, but I'd like to get your review, so I can fix issues before dev-cut.
Take your time. I'll be on holiday till EOW.

ewolinetz commented:
So far, so good. I think you're missing changes in `filter_plugins/oo_filters.py` too. They're more apparent in https://github.com/openshift/openshift-ansible/pull/5176

sdodson commented:
/assign ewolinetz

smarterclayton commented on filter_plugins/oo_filters.py:
```diff
@@ -985,6 +997,18 @@ def oo_persistent_volume_claims(hostvars, persistent_volume_claims=None):
         persistent_volume_claim = oo_component_pv_claims(hostvars, 'metrics')
         if persistent_volume_claim is not None:
             persistent_volume_claims.append(persistent_volume_claim)
+    if 'prometheus' in hostvars['openshift']:
```
@sdodson this sort of duplication in core modules is exactly what my underlying concern is with the coupling between our different components.  Having to do this is a code smell and is coupling completely unrelated to playbooks.

smarterclayton commented on roles/openshift_storage_nfs/tasks/main.yml:
```diff
@@ -35,6 +35,9 @@
     - "{{ openshift.logging }}"
     - "{{ openshift.loggingops }}"
     - "{{ openshift.hosted.etcd }}"
+    - "{{ openshift.prometheus }}"
```
The fact that this role knows about all possible things that could ever use NFS is a serious architectural coupling problem.

(zohar not picking on you, picking on where we need to change course in ansible)

ewolinetz commented:
/test tox

ewolinetz commented:
@zgalor looks like all the tests passed. Do you want to remove the WIP in the title and squash? Then we can give it a lgtm

zgalor commented:
@ewolinetz tempting... But I want to run some more tests and also update the examples, and then I'll remove the WIP

sdodson commented:
@zgalor still more testing you'd like to do before we accept this?

zgalor commented:
@sdodson this is not ready yet. I need to add documentation, and some more testing.
I'm on PTO this week so it won't advance. I'll remove the wip next week and tag you once done

sdodson commented:
Ok thanks.

zgalor commented:
@sdodson @ewolinetz now I finally tested everything, squashed and removed the WIP.
Please re-review and if all tests pass you can merge.
Not much changed since your last review. Just rebased, added examples and updated the image tag for alertmanager.

tbielawa commented on inventory/byo/hosts.example:
```diff
@@ -615,6 +615,71 @@ openshift_master_identity_providers=[{'name': 'htpasswd_auth', 'login': 'true',
+#openshift_prometheus_storage_access_modes=['ReadWriteOnce']
+#openshift_prometheus_storage_nfs_directory=/exports
+#openshift_prometheus_storage_nfs_options='*(rw,root_squash)'
+#openshift_prometheus_storage_volume_name=prometheus
```
README requires variable documentation updates to describe all the new config parameters 

If some of these are not intended to be overridden by the users (e.g., *constants*) then they should be moved into the `vars/` directory and may not require additional documentation.

zgalor replied:
@tbielawa I started changing the readme but then I realized these variables are not in this role scope, and shouldn't be added in the role README.
If any, this README should update:
https://github.com/openshift/openshift-ansible/tree/master/roles/openshift_storage_nfs
but, it currently doesn't have all the logging, metrics, etc vars in it.

Was that the one you meant?

sdodson commented:
We should port this over to using the openshift_nfs role in the future but accept it for now.

sdodson commented:
/lgtm

sdodson commented:
/lgtm cancel

tbielawa commented:
@sdodson and I just spoke, I'm going to lgtm this again so long as we can get the documentation addressed asap in a follow-up PR.

tbielawa commented:
/lgtm

sdodson commented:
/retest

On Mon, Oct 16, 2017 at 3:24 PM, OpenShift CI Robot <
notifications@github.com> wrote:

> @zgalor <https://github.com/zgalor>: The following test *failed*, say
> /retest to rerun them all:
> Test name Commit Details Rerun command
> ci/openshift-jenkins/upgrade 418b742
> <https://github.com/openshift/openshift-ansible/commit/418b742c365ef8dce4e14f9486ea658495029df3>
> link
> <https://openshift-gce-devel.appspot.com/build/origin-ci-test/pr-logs/pull/openshift_openshift-ansible/5459/test_pull_request_openshift_ansible_extended_conformance_install_update/2505/> /test
> upgrade
>
> Full PR test history
> <https://openshift-gce-devel.appspot.com/pr/openshift_openshift-ansible/5459>.
> Your PR dashboard <https://openshift-gce-devel.appspot.com/pr/zgalor>.
> Please help us cut down on flakes by linking to
> <https://github.com/kubernetes/community/blob/master/contributors/devel/flaky-tests.md#filing-issues-for-flaky-tests>
> an open issue
> <https://github.com/openshift/openshift-ansible/issues?q=is:issue+is:open>
> when you hit one in your PR.
>
> Instructions for interacting with me using PR comments are available here
> <https://github.com/kubernetes/community/blob/master/contributors/devel/pull-requests.md>.
> If you have questions or suggestions related to my behavior, please file an
> issue against the kubernetes/test-infra
> <https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:>
> repository. I understand the commands that are listed here
> <https://github.com/kubernetes/test-infra/blob/master/commands.md>.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/openshift/openshift-ansible/pull/5459#issuecomment-337005980>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAC8Idwv--2E8M7AnvCcHzHn2_cjYh1Pks5ss61QgaJpZM4PcVBX>
> .
>

zgalor commented:
/retest

zgalor commented:
/retest

zgalor commented:
/retest

zgalor commented:
/test logging
